### PR TITLE
allowing materials.xml to be specified

### PR DESCRIPTION
**Pull request by shimwell**
The ```Results.export_to_materials``` method in ```openmc.deplete``` is super useful.

This PR is an attempt to make it a tiny bit more flexible by allowing the path to the materials.xml to be specified.

This is useful when running multiple depletion steps and wanting to get the materials for a seperate folder or elsewhere.

It is useful to have the different stages stored in separate file locations so that the materials at each time step can be investigated.

Would you be able to review @eepeterson as this is similar to changes you had in your R2SModel branch but I've tried to use the path argument with PathLike the same way it is used elsewhere in openmc (e.g. [here](https://github.com/openmc-dev/openmc/blob/3f5b90042c378f4182ef196ac075dd8cd8cec686/openmc/material.py#L1587))

### CODE REVIEW

**Comment by eepeterson** [comment]
Thanks @shimwell! This is indeed helpful for those workflows.

**Review by eepeterson — APPROVED** [review]

File: openmc/deplete/results.py [hunk 1]
```diff
@@ -394,6 +395,8 @@ def export_to_materials(
             If not provided, nuclides from the cross_sections element of
             materials.xml will be used. If that element is not present,
             nuclides from openmc.config['cross_sections'] will be used.
+        path : PathLike
+            Path to file to write. Defaults to 'materials.xml'.
```
Comment (paulromano) [inline openmc/deplete/results.py:399]: The method below doesn't write to materials.xml. Instead, it uses an _existing_ materials.xml as the basis for generating an updated `Materials` object. The description here needs to be changed to reflect that.

**Review by paulromano — CHANGES_REQUESTED** [review]

File: openmc/deplete/results.py (hunk 1 above)
Comment (shimwell) [inline openmc/deplete/results.py:399]:
```suggestion
            Path to materials XML file to read. Defaults to 'materials.xml'.
```

**Review by paulromano — APPROVED** [review]

**Comment by shimwell** [comment]
I have merged in the latest develop branch (which fixes the CI) and am now rerunning the tests as I think that a pass is required to merge this PR

**Comment by shimwell** [comment]
Super tests are passing, I think this is ready to merge

**Comment by paulromano** [comment]
Thanks @shimwell. For future reference, you don't actually need to merge in develop to your branch. When CI runs, it is actually testing a merge commit, so just retriggering CI once the `develop` branch is updated should be sufficient.

**Pull request merged**